### ISSUE TITLE
Solving vars with the current value changes the behavior

### ISSUE THREAD

**Issue by vitaly-eureka-security** [diegoholiveira/jsonlogic#81]
In the playground (https://jsonlogic.com/play.html)  and in version < 3.5.1 the following rule
```
{
      "some": [
        {"var": "A"},
        {"!=": [
          {"var": ".B"},
          {"var": "B"}
        ]}
      ]
}
```
when applied to 
```
{"A":[{"B":1}], "B":2}
```
returns `true`.

However, since this commit https://github.com/diegoholiveira/jsonlogic/commit/0926afe273c8480f0bd1c63eb2ab59564cb80e71, the return value is false.

Is it a desired change? If so, how the previous functionality (of comparing the value with the value from external scope) can be achieved.

**Comment by diegoholiveira** [comment]
Not a desirable change. I'll take a look.

**Comment by diegoholiveira** [comment]
Fixed with #82
